typescript/deploy/src/utils.ts

```diff
@@ -0,0 +1,153 @@
+import yargs from 'yargs';
+import { ethers } from 'ethers';
+import { NonceManager } from '@ethersproject/experimental';
+import {
+  AbacusCore,
+  ChainName,
+  domains,
+  MultiProvider,
+} from '@abacus-network/sdk';
+import { types } from '@abacus-network/utils';
+import { TransactionConfig, EnvironmentConfig } from './config';
+import { RouterConfig } from './router';
+
+export function getArgs() {
+  return yargs(process.argv.slice(2))
+    .alias('e', 'env')
+    .describe('e', 'deploy environment')
+    .help('h')
+    .alias('h', 'help');
+}
+
+export async function importModule(moduleName: string): Promise<any> {
+  const importedModule = await import(moduleName);
+  return importedModule;
+}
+
+export async function getEnvironmentConfig<E extends EnvironmentConfig>(
+  moduleName: string,
+): Promise<E> {
+  return importModule(moduleName);
+}
+
+export async function getEnvironment(): Promise<string> {
+  return (await getArgs().argv).e as Promise<string>;
+}
+
+export function getRouterConfig(core: AbacusCore): RouterConfig {
+  const xAppConnectionManager: Record<string, types.Address> = {};
+  core.domainNames.map((name) => {
+    const contracts = core.mustGetContracts(name);
+    xAppConnectionManager[name] = contracts.xAppConnectionManager.address;
+  });
```

Comment:
another place I think we can more strongly type
could also be a function on `AbacusCore` `getConnectionManagers` that returns this